modules/library/src/main/scala/zio/elasticsearch/query/FunctionScoreFunction.scala

```diff
@@ -0,0 +1,729 @@
+   * [[zio.elasticsearch.query.ScriptScoreFunction]] function allows you to wrap another query and customize the scoring
+   * of it optionally with a computation derived from other numeric field values in the doc using a script expression.
+   *
+   * @param script
+   *   the [[zio.elasticsearch.script.Script]] that will be used to calculate score
+   * @return
+   *   an instance of [[zio.elasticsearch.query.ScriptScoreFunction]] that represents the function score query to be
+   *   performed.
+   */
+  def scriptScoreFunction(script: Script): ScriptScoreFunction[Any] =
+    ScriptScoreFunction[Any](script = script, weight = None, filter = None)
+
+  /**
+   * Constructs an instance of [[zio.elasticsearch.query.WeightFunction]] using the specified parameters.
+   * [[zio.elasticsearch.query.WeightFunction]] score allows you to multiply the score by the provided weight.
+   *
+   * @param weight
+   *   the number you wish to multiply the score with
+   * @return
+   *   an instance of [[zio.elasticsearch.query.WeightFunction]] that represents the function score query to be
+   *   performed.
+   */
+  def weightFunction(weight: Double): WeightFunction[Any] =
+    WeightFunction[Any](weight = weight, filter = None)
+
+}
+
+final case class DecayFunction[S](
+  field: String,
+  decayFunctionType: DecayFunctionType,
+  origin: String,
+  scale: String,
+  decay: Option[Double],
+  filter: Option[ElasticQuery[S]],
+  multiValueMode: Option[MultiValueMode],
+  offset: Option[String],
+  weight: Option[Double]
+) extends FunctionScoreFunction[S] { self =>
+
+  /**
+   * Sets the `decay` parameter for the [[zio.elasticsearch.query.DecayFunction]]. It defines how documents are scored
+   * at the distance given at scale. If no decay is defined, documents at the distance scale will be scored 0.5.
+   *
+   * @param value
+   *   the [[Double]] value for `decay` parameter
+   * @return
+   *   a new instance of the [[zio.elasticsearch.query.DecayFunction]] enriched with the `decay` parameter.
+   */
+  def decay(value: Double): DecayFunction[S] =
+    self.copy(decay = Some(value))
+
+  def filter(filter: ElasticQuery[Any]): FunctionScoreFunction[S] =
+    self.copy(filter = Some(filter))
+
+  def filter[S1 <: S: Schema](filter: ElasticQuery[S1]): DecayFunction[S1] =
+    DecayFunction[S1](
+      field = self.field,
+      decayFunctionType = self.decayFunctionType,
+      origin = self.origin,
+      scale = self.scale,
+      decay = self.decay,
+      filter = Some(filter),
+      multiValueMode = self.multiValueMode,
+      offset = self.offset,
+      weight = self.weight
+    )
+
+  /**
+   * Sets the `multiValueMode` parameter for the [[zio.elasticsearch.query.DecayFunction]]. If a field used for
+   * computing the decay contains multiple values, per default the value closest to the origin is chosen for determining
+   * the distance. This can be changed by setting `multiValueMode`.
```

Comment:
Maybe consider something like this:
This parameter is used for changing of ...?